tests/test_deposit.py

```diff
@@ -84,3 +86,34 @@ def test_deposit_updates_total_deposits(casper, funded_privkey, deposit_amount,
 
     assert casper.total_curdyn_deposits_in_wei() == deposit_amount
     assert casper.total_prevdyn_deposits_in_wei() == deposit_amount
+
+
+@pytest.mark.parametrize(
+    'warm_up_period,epoch_length',
+    [
+        (10, 5),
+        (25, 10),
+        (100, 50),
+    ]
+)
+def test_deposit_during_warm_up_period(casper_chain, casper, funded_privkey, deposit_amount,
+                                       deposit_validator, new_epoch, warm_up_period, epoch_length):
+    validator_index = deposit_validator(funded_privkey, deposit_amount)
+
+    expected_start_dynasty = casper.dynasty() + 2
+    assert casper.validators__start_dynasty(validator_index) == expected_start_dynasty
+
+    new_epoch()  # new_epoch mines through warm_up_period on first call
```

Comment:
If I understand correctly, as described in #141 `initialize_epoch` can be called after the `WARM_UP_PERIOD` right? 
why `new_epoch`can be called before the expected start dynasty?

Reply:
`new_epoch` actually encapsulates walking through the warm up period if there is a warm up period.

This functionality of the `new_epoch` fixture it tested [here](https://github.com/ethereum/casper/pull/142/files#diff-d2d5f544b29602ba7340415d252e40b6R14) and then used throughout.